File: jfr-agent/src/main/java/sk/fiit/dp/model/ExecutionSample.java
package sk.fiit.dp.model;

import java.util.List;

public class ExecutionSample {
    public long timestamp;
    public String threadName;
    public List<Frame> frames;

    public ExecutionSample(long timestamp, String threadName, List<Frame> frames) {
        this.timestamp = timestamp;
        this.threadName = threadName;
        this.frames = frames;
    }

    public static class Frame {
        public String className;
        public String method;
        public int line;

        public Frame(String className, String method, int line) {
            this.className = className;
            this.method = method;
            this.line = line;
        }
    }
}

File: src/main/kotlin/sk/fiit/dp/dpintellijplugin/data/project/ExecutionSample.kt
package sk.fiit.dp.dpintellijplugin.data.project

data class ExecutionSample(
    val timestamp: Long,
    val threadName: String,
    val frames: List<Frame>
)

data class Frame(
    val className: String,
    val method: String,
    val line: Int,
)